test-data/unit/check-literal.test

```diff
@@ -2038,6 +2038,29 @@ del test[bad_keys]              # E: Key "a" of TypedDict "Test" cannot be delet
 [typing fixtures/typing-typeddict.pyi]
 [out]
 
+[case testTypedDictLiteralTypeSetDefault]
+from typing_extensions import Literal
+from mypy_extensions import TypedDict
+from typing import List, Optional
+
+class A: pass
+
+class Test(TypedDict):
+    a: List[Optional[int]]
+    
+test: Test
+
+a: Literal["a"]
+b: Literal["b"]
```

Comment:
Try using a different name for the variable compared to the literal string value, e.g. `aa: Literal["a"]`. This will expose an issue with your code, I believe.